File: home/dot_local/share/lazyvim/capabilities/node.mjs
import { captureCommandOutput } from "../lib/commands.mjs";
import { defineCapability } from "./contract.mjs";

export default defineCapability({
  id: "language.node",
  requires: ["foundation"],
  setup({ platform }) {
    platform.configureNodeHost();
  },
  verify({ platform, versions }) {
    platform.verifyNodeHost();
    const actual = captureCommandOutput("node", ["--version"]);
    if (actual !== `v${versions.node}`) {
      throw new Error(`Expected managed Node v${versions.node}, got ${actual}`);
    }
  },
});

File: home/dot_local/share/lazyvim/lua/lazyvim_capabilities/capabilities/node.lua
local commands = require("lazyvim_capabilities.commands")
local define = require("lazyvim_capabilities.contract")
return define({
	id = "language.node",
	requires = { "foundation" },
	setup = function(context)
		context.platform.configure_node()
	end,
	verify = function(context)
		context.platform.verify_node()
		local actual = commands.capture("node", { "--version" })
		assert(
			actual == "v" .. context.versions.node,
			("Expected managed Node v%s, got %s"):format(context.versions.node, actual)
		)
	end,
})
